iodata/overlap.py

```diff
@@ -156,6 +214,17 @@ def gob_cart_normalization(alpha: np.ndarray, n: np.ndarray) -> np.ndarray:
         The normalization constant for the gaussian cartesian basis.
 
     """
-    vfac2 = np.vectorize(factorialk)
-    return np.sqrt((4 * alpha)**sum(n) * (2 * alpha / np.pi)**1.5
-                   / np.prod(vfac2(2 * n - 1, 2)))
+    vfac2 = np.vectorize(factorial2)
+    return np.sqrt((4 * alpha) ** sum(n) * (2 * alpha / np.pi) ** 1.5
+                   / np.prod(vfac2(2 * n - 1)))
+
+
+def iter_cart_alphabet2(n):
```

Comment:
Would it be possible to reuse the function in `iodata/obasis.py`? See https://github.com/theochem/iodata/blob/master/iodata/basis.py#L318

Reply:
I will look more into it, but when I tried it did not work since that produces an iterable, and I needed to be able to pass explicit arrays to the vectorized function. I may have simply implemented it wrong.

Reply:
Would the following be ok?

```python
>>> import numpy as np
>>> from iodata.basis import iter_cart_alphabet
>>> np.array(list(iter_cart_alphabet(3)))
array([[3, 0, 0],
       [2, 1, 0],
       [2, 0, 1],
       [1, 2, 0],
       [1, 1, 1],
       [1, 0, 2],
       [0, 3, 0],
       [0, 2, 1],
       [0, 1, 2],
       [0, 0, 3]])
```

Reply:
Yes, that is a much more pythonic way. I have changed it.